From 6e4672f00c3bc94e8be5bc7878bc6628ad6931ca Mon Sep 17 00:00:00 2001
From: Nic Boet <nic@boet.cc>
Date: Thu, 1 Oct 2026 23:10:21 -0500
Subject: [PATCH] Fix segfault with -f when TERM is unset

The -f handling passed getenv("TERM") straight to strcmp(), which
crashes when TERM is not set. Treat an unset TERM like any other
non-"linux" value and set it to "linux".

Fixes #174. (Backport to 2.0 of 0d29e84; upstream PR #220.)
---
 cmatrix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmatrix.c b/cmatrix.c
index 15c9451..51ccb44 100644
--- a/cmatrix.c
+++ b/cmatrix.c
@@ -373,7 +373,7 @@ int main(int argc, char *argv[]) {
         }
     }
 
-    if (force && strcmp("linux", getenv("TERM"))) {
+    if (force && (getenv("TERM") == NULL || strcmp("linux", getenv("TERM")))) {
         /* setenv is much more safe to use than putenv */
         setenv("TERM", "linux", 1);
     }
